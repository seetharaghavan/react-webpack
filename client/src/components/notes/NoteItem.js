import React from 'react'; 
import { useDispatch } from 'react-redux';
import { deleteNote } from '../../services/actions';
import { SELECT_NOTE, SIDEBAR_TOGGLE } from '../../types';

const NoteItem = ({ noteInfo }) => {
  const { title, body } = noteInfo;
  const dispatch = useDispatch();
  return (
    <li className="note-item">
      <div className="note-holder">
        <div
          className="note-info content"
          onClick={() => {
            dispatch({ type: SELECT_NOTE, payload: noteInfo }); 
            dispatch({ type: SIDEBAR_TOGGLE, payload: false}); 
          }}
        >
          <h5 className="note-title text">{title}</h5>
          <p className="note-description text">{body}</p>
        </div>
        <div>
          <button
            className="btn btn-light"
            onClick={() => dispatch(deleteNote(noteInfo))}
          >
            x
          </button>
        </div>
      </div>
    </li>
  );
};

export default NoteItem; 
