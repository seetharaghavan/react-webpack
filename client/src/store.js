import {createStore, combineReducers, applyMiddleware} from 'redux'; 
import ReduxThunk  from 'redux-thunk'; 
import authReducer from './reducers/auth.reducer';
import noteReducer from './reducers/note.reducer';
import sideBarReducer from './reducers/sidebar.reducer';

const store = createStore( 
    combineReducers({
        authState: authReducer,
        noteState: noteReducer,
        sidebarState: sideBarReducer
    }),
    {},
    applyMiddleware(ReduxThunk)
)

export default store; 

