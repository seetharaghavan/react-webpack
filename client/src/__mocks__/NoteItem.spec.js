import React from "react";
import renderer from "react-test-renderer";
import localStorage from "./localstorage";
import { Provider } from "react-redux";
import store from "../store";
import NoteItem from "../components/notes/NoteItem";
import { render, fireEvent, cleanup } from "@testing-library/react";

afterEach(cleanup);

describe("Note UI Test", () => {
  it("should render without crash", async () => {
    let notesItem = {
      title: "Test Body",
      body: "Test Body",
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <NoteItem noteInfo={notesItem} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render title and body", async () => {
    let notesItem = {
      title: "Test Title",
      body: "Test Body",
    };
    let { getByText } = render(
      <Provider store={store}>
        <NoteItem noteInfo={notesItem} />
      </Provider>
    );
    expect(getByText('Test Title')).toBeInTheDocument();
    expect(getByText('Test Body')).toBeInTheDocument();
  });
});
