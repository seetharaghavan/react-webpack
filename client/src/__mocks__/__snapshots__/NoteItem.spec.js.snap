// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note UI Test should render without crash 1`] = `
<li
  className="note-item"
>
  <div
    className="note-holder"
  >
    <div
      className="note-info content"
      onClick={[Function]}
    >
      <h5
        className="note-title text"
      >
        Test Body
      </h5>
      <p
        className="note-description text"
      >
        Test Body
      </p>
    </div>
    <div>
      <button
        className="btn btn-light"
        onClick={[Function]}
      >
        x
      </button>
    </div>
  </div>
</li>
`;
